packages/tupaia-web/src/features/Map/MapOverlaySelector/DesktopMapOverlaySelector.tsx

```diff
@@ -146,47 +164,59 @@ export const DesktopMapOverlaySelector = ({
 }: DesktopMapOverlaySelectorProps) => {
   const { projectCode, entityCode } = useParams();
   const { hasMapOverlays } = useMapOverlays(projectCode, entityCode);
-  const { period } = useActiveMapOverlayReport();
-
+  const { data: mapOverlayData } = useMapOverlayReport();
+  const [mapModalOpen, setMapModalOpen] = useState(false);
+  const handleOpen = () => {
+    setMapModalOpen(true);
+  };
   return (
-    <Wrapper>
-      <Header>
-        <Heading>Map Overlays</Heading>
-      </Header>
-      <Container>
-        <TitleWrapper>
-          <MapOverlaySelectorTitle />
-          <MapOverlayDatePicker />
-        </TitleWrapper>
-        {hasMapOverlays && (
-          <OverlayLibraryAccordion
-            expanded={overlayLibraryOpen}
-            onChange={toggleOverlayLibrary}
-            square
-          >
-            <OverlayLibraryHeader
-              expandIcon={<ArrowDropDown />}
-              aria-controls="overlay-library-content"
-              id="overlay-library-header"
+    <>
+      {mapModalOpen ? <MapTableModal setIsOpen={setMapModalOpen} /> : null}
+      <Wrapper>
+        <Header>
+          <Heading>Map Overlays</Heading>
+          <MapTableButton>
+            <Tooltip arrow interactive placement="top" title="Generate Report">
+              <TableAssignmentIcon onClick={() => handleOpen()} />
```

Comment:
This doesn't need to be an arrow function. When we have arrow functions inline, it technically has a slight performance impact, because it causes re-rendering of the component.